watchtower-plugin/src/retrier.rs

```diff
@@ -64,18 +64,22 @@ impl RetryManager {
                     self.add_pending_appointment(tower_id, locator);
                 }
                 Err(TryRecvError::Empty) => {
-                    // Keep only running retriers and retriers ready to be re-started.
+                    // Keep only running retriers and retriers ready to be started/re-started.
                     // This will remove failed ones and ones finished successfully and have no pending appointments.
                     self.retriers
-                        .retain(|_, retrier| retrier.is_running() || retrier.ready());
+                        .retain(|_, retrier| retrier.is_running() || retrier.should_start());
                     // Start all the ready retriers.
                     for retrier in self.retriers.values() {
-                        if retrier.ready() {
+                        if retrier.should_start() {
                             self.start_retrying(retrier);
                         }
                     }
-                    // Sleep to not waste a lot of CPU cycles.
-                    tokio::time::sleep(Duration::from_secs(1)).await;
+                    // When max elapsed time is zero, this is probably a test environment and we want
+                    // things to be really fast. So don't sleep.
+                    if self.max_elapsed_time_secs != 0 {
+                        // Sleep to not waste a lot of CPU cycles.
+                        tokio::time::sleep(Duration::from_secs(1)).await;
+                    }
```

Comment:
This part should let the test pass. I hope.
But the retrier thread will do a lot of busy waiting when ran by tests, which isn't ideal. The best idea I have is to make this one second sleep a variable that we can control from tests and account for it (e.g. in the tests we set it to 1 and sleep 1 sec to account for it).

Reply:
Didn't fix it: https://github.com/meryacine/rust-teos/actions/runs/3066853645/jobs/4952503002#step:9:6925

Reply:
So if I'm not mistaken the issue here is that the retrier is not flagging the tower as unreachable fast enough, is it not? Basically we are checking the state before the retrier has given up. Given the retrier is set to give up "immediately" (`"watchtower-max-retry-time": 0`) we can just add a small sleep in the test after sending the payment.

Reply:
Yeah a small sleep should fix it, I was trying to avoid it though.
I will try to make that one second sleep settable from the command line so tests doesn't depend on other code.

Reply:
Even though it's not elegant, it may still be necessary. Otherwise, the test may randomly fail. It's basically a race between the retrier failing and the test checking the tower state.